### PR TITLE
Upgrade JFreeChart 

### LINKED ISSUE #26: Upgrade to version 1.5
Some time ago version 1.5 has been released; it is not a 1:1 replacement as there API changes.

### PR DESCRIPTION
**Pull request by dizzzz**
fixes #26

### CODE REVIEW

**Comment by dizzzz** [comment]
[JFreechart copy.pdf](https://github.com/eXist-db/jfreechart/files/3282954/JFreechart.copy.pdf)

File: src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java [hunk 1]
```diff
@@ -145,13 +145,13 @@ public static JFreeChart createJFreeChart(final String chartType, final Configur
...
                 break;
 
-            case "BarChart3D":
-                chart = ChartFactory.createBarChart3D(
-                        conf.getTitle(), conf.getCategoryAxisLabel(), conf.getValueAxisLabel(), categoryDataset,
-                        conf.getOrientation(), conf.isGenerateLegend(), conf.isGenerateTooltips(), conf.isGenerateUrls());
-
-                setCategoryChartParameters(chart, conf);
-                break;
+//            case "BarChart3D":
```
Comment (adamretter) [inline src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java]: Are these various chart types meant to be commented out?

File: src/test/java/org/exist/xquery/modules/jfreechart/ExampleModuleTest.java [hunk 2]
```diff
@@ -32,12 +12,12 @@
...
 
-    private Sequence executeQuery(final String xquery) throws EXistException, PermissionDeniedException, XPathException {
-        final BrokerPool pool = existEmbeddedServer.getBrokerPool();
-        final XQuery xqueryService = pool.getXQueryService();
-
-        try(final DBBroker broker = pool.get(Optional.of(pool.getSecurityManager().getSystemSubject()))) {
-            return xqueryService.execute(broker, xquery, null);
-        }
-    }
+//    private Sequence executeQuery(final String xquery) throws EXistException, PermissionDeniedException, XPathException {
```
Comment (adamretter) [inline src/test/java/org/exist/xquery/modules/jfreechart/ExampleModuleTest.java]: Tests seem to be commented out?

**Review by adamretter — COMMENTED** [review]
Tests seem to have been commented out in ExampleModuleTest and some chart types seem to have been commented out too?

**Comment by dizzzz** [comment]
right, I will make that consistent.

File: src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java (hunk 1 above)
Comment (dizzzz) [reply src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java]: yes, as they are removed from v1.5 ; the non 3d ones look much better now (according to some reports)

File: src/test/java/org/exist/xquery/modules/jfreechart/ExampleModuleTest.java (hunk 2 above)
Comment (dizzzz) [reply src/test/java/org/exist/xquery/modules/jfreechart/ExampleModuleTest.java]: yea nothing happened here. I will remove the test completely.

File: src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java (hunk 1 above)
Comment (adamretter) [reply src/main/java/org/exist/xquery/modules/jfreechart/JFreeChartFactory.java]: Okay - maybe just remove that code then!

**Comment by joewiz** [comment]
Fantastic, Dannes!

**Pull request merged**